packages/plugins/src/$RefPlugin.ts

```diff
@@ -139,7 +140,7 @@ const $RefPlugin: PluginType<RefsPluginPage, unknown, RefsPluginConfigData> = {
           pageExtensions,
           ignorePages
         );
-        normalisedRefs[fullPath] = { ...page, ...expandedRefs };
+        normalisedRefs[fullPath] = deepmerge(page, expandedRefs);
```

Comment:
Without this, we were missing info that may be included inside the `data` metadata of a page BUT is not part of a ref.  The merge here seems to fix the issue of dropping non-ref'd data while still preventing the circular ref issue encountered when we were using lodash merge.  Will continue to monitor the behaviour of the ref plugin.